### DISCUSSION TITLE
How to recover from a revoked install token?

### DISCUSSION THREAD

**Discussion by don-smith** [asciinema/discussions#37] [Q&A]
I revoked an install token because I wanted to associate my asciinema installation with a different email address/account. In retrospect, I should have just changed the username and email address in my existing account, but alas, I revoked the install token.

Now each time I run `asciinema auth` it keeps using the same initial token. I've tried deleting `~/.config/asciinema/install-id` and it doesn't seem to recreate it. I manually recreated the `install-id` file with a different token value and appended the new value to `https://asciinema.org/connect/` but `asciinema upload` still fails because (I assume) it's still using the original (now revoked) token rather than the value in the `install-id` file. I also uninstalled and re-installed asciinema but the same token value keeps being used.

How do I reset asciinema or get it to use a different token value?

If this isn't possible using the CLI and asciinema.org and you need to do something on the backend, the token I'd like unrevoked is associated with the https://asciinema.org/~antiantonym (don@antiantonym.com) account.

**Comment by don-smith** [comment]
After poking through the asciinema code base, I think I have a work-around. I've set the `ASCIINEMA_INSTALL_ID` environment variable to the same custom value I've placed in `~/.config/asciinema/install-id` and after associating that token with my preferred account, `asciinema upload` is using the new value.

If there is another way to do this so I don't have to use the environment variable, please let me know.

Thank you, this is such a cool project!

**Comment by ku1ik** [comment]
Which version of asciinema you have? install-id file is only in 2.0, previous versions kept it in the config file as `token` in `[api]` section.

**Comment by LegNeato** [comment]
FWIW in [asciinema-rs](https://github.com/LegNeato/asciinema-rs) we added an `--install-id` flag to the `auth` subcommand. Might be worthwhile to add it to the python version too.

**Comment by ku1ik** [comment]
@LegNeato what does it do? Generates new one?

**Comment by LegNeato** [comment]
@sickill it uses it for that session and writes it out to the install-id file. Basically let's you set it if you already have one

**Comment by ku1ik** [comment]
@don-smith maybe information from this FAQ entry could be helpful: https://discourse.asciinema.org/t/why-upload-failed-invalid-or-revoked-install-id/27
